sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_client.py

```diff
@@ -60,10 +60,12 @@ def _build_auth(credential):
         auth['resourceTokens'] = credential  # type: ignore
     elif hasattr(credential, '__iter__'):
         auth['permissionFeed'] = credential
+    elif hasattr(credential, 'get_token'):
+        auth['clientSecretCredential'] = credential
     else:
         raise TypeError(
-            "Unrecognized credential type. Please supply the master key as str, "
-            "or a dictionary or resource tokens, or a list of permissions.")
+            "Unrecognized credential type. Please supply the master key as a string "
+            "or a dictionary, or resource tokens, or a list of permissions, or a ClientSecretCredential.")
```

Comment:
ClientSecretCredential -> any instance of a class implementing TokenCredential (see azure.identity for specific implementations such as ClientSecretCredential)